File: src/DashboardUI/src/data-contracts/EstimateConsumptiveUseApplicantRequest.ts
import { ApplicationStoreRequestBase } from './ApplicationStoreRequestBase';
import { CompensationRateUnits } from './CompensationRateUnits';
import { MapPolygon } from './MapPolygon';

export interface ApplicantEstimateConsumptiveUseRequest extends ApplicationStoreRequestBase {
  $type: 'ApplicantEstimateConsumptiveUseRequest';
  waterConservationApplicationId: string;
  waterRightNativeId: string;
  polygons: MapPolygon[];
  compensationRateDollars: number | undefined;
  units: Exclude<CompensationRateUnits, CompensationRateUnits.None> | undefined;
}

File: src/API/WesternStatesWater.WestDaat.Contracts.Client/Requests/Conservation/EstimateConsumptiveUseApplicantRequest.cs
using WesternStatesWater.WestDaat.Common.DataContracts;

namespace WesternStatesWater.WestDaat.Contracts.Client.Requests.Conservation;

public class EstimateConsumptiveUseApplicantRequest : ApplicationStoreRequestBase
{
    public Guid WaterConservationApplicationId { get; set; }

    public string WaterRightNativeId { get; set; } = null!;

    public MapPolygon[] Polygons { get; set; }

    public int? CompensationRateDollars { get; set; }

    public CompensationRateUnits? Units { get; set; }
}
